service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceMetadataManager.java

```diff
@@ -86,17 +98,66 @@ public ControlledResource validateControlledResourceAndAction(
       AuthenticatedUserRequest userRequest, UUID workspaceUuid, UUID resourceId, String action) {
     stageService.assertMcWorkspace(workspaceUuid, action);
     WsmResource resource = resourceDao.getResource(workspaceUuid, resourceId);
-
     ControlledResource controlledResource = resource.castToControlledResource();
-    SamRethrow.onInterrupted(
-        () ->
-            samService.checkAuthz(
-                userRequest,
-                controlledResource.getCategory().getSamResourceName(),
-                resourceId.toString(),
-                action),
-        "checkAuthz");
-    return controlledResource;
+    String samName = controlledResource.getCategory().getSamResourceName();
+
+    // Every case exits the validation
+    switch (controlledResource.getState()) {
+      case READY:
+        // Primary success case: check auth and return the resource
+        SamRethrow.onInterrupted(
+            () -> samService.checkAuthz(userRequest, samName, resourceId.toString(), action),
+            "checkAuthz");
+        return controlledResource;
+
+      case CREATING, DELETING, UPDATING:
+        // Check forbidden before throwing the busy resource error
+        SamRethrow.onInterrupted(
+            () -> samService.checkAuthz(userRequest, samName, resourceId.toString(), action),
+            "checkAuthz");
+        throw new ResourceIsBusyException(
+            "Another operation is running on the resource; wait and try again");
+
+      case BROKEN_DELETE:
+        // If the resource is in the BROKEN_DELETING state, then it is in an unknown broken state
+        // and cannot be deleted by a user. It has to be cleaned up manually by Terra personnel.
+        throw new ResourceIsCorruptException(
+            "Resource is in an internally corrupted state. Contact support.");
+
+      case BROKEN:
+        // If the resource is in the BROKEN state, then there may be no Sam resource. We do our
```

Comment:
Delete is the only operation allowed on BROKEN. No operations are allowed on BROKEN_DELETE. It requires manual intervention. It is the state for dismal failures on the delete path. I don't think any current code can arrive in BROKEN_DELETE because the flight will stop with a dismal failure. My (unimplemented) idea is that we can alter the undo steps to mark a failed undo (in the working map) and successfully continue. The final step would leave the metadata in the database and mark it BROKEN_DELETE.

That said, I am uncertain it is the right design. On the one hand, it makes it clear to the UI user that there is a problem, and provides some explanation. Otherwise, the object is stuck in DELETING state with nothing happening. On the other hand, suppose the failure was some outage that is now restored. We could use `stairctl` to for the flight state to READY and allow the undo to complete normally. Which is better than manually going in to Sam or the cloud to delete resources.

I'll give this issue more visibility. Thanks for the prompt.